.github/workflows/build-pr.yml

```diff
@@ -197,10 +207,22 @@ jobs:
     name: ubuntu-i386-(${{ matrix.build_type.type }}, ${{ matrix.shared_libs.name }}, ${{ matrix.with_openssl.name }})
 
     steps:
+      - name: Checkout Code
+        uses: actions/checkout@v1
+        with:
+          ref: ${{ needs.get-refs.outputs.ref }}
+          token: ${{ secrets.GH_TOKEN }}
+
+      - name: Read Config
+        run: cat .github/config.env >> $GITHUB_ENV
+
+      # Install Java via apt, can't use `setup-java` as the runner image is too old to support a compatible Node installation
+      # Newest available is Node 8, but Node 20 is required
+      # https://github.com/actions/setup-node/issues/922
       - name: Install Necessary Packages
         run: |
           apt-get update
-          apt-get install -y build-essential cmake curl git libssl-dev maven net-tools openjdk-11-jre-headless gdb curl
+          apt-get install -y build-essential cmake curl git libssl-dev net-tools openjdk-${{ env.JAVA_VERSION }}-jre-headless gdb curl
```

Comment:
why install Node?

Reply:
[The `setup-java` action is written in TypeScript](https://github.com/actions/setup-java) and executes on the Node installation that's bundled with GitHub runners.
This image does have Node, but the version is too old and there's nothing newer.

Reply:
Can you put this explanation into the comment so that we remember this?

Reply:
> Can you put this explanation into the comment so that we remember this?

c018d778f0fbf6edb4183fe46127992e8705ffba

Hopefully that's clearer? Let me know if not :)